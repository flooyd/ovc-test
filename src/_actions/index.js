const addUsers = users => ({
  type: 'ADD_USERS',
  users
})

const fetchUsers = () => {
  return fetch('https://jsonplaceholder.typicode.com/users');
}

const getUsers = () => {
  return function(dispatch) {
    return fetchUsers()
      .then(response => {
        return response.json();
      })
      .then(users => {
        dispatch(addUsers(users))
      })
      .catch(err => {
        console.log(err);
      })
  }
}

export default getUsers;
