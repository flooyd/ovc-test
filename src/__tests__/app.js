import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App'
import store from '../_store'
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux';
import {getUsers} from '../_actions';

/*I am fairly new to testing async action creators with jest
and I know the below approach probably isn't the best way to
go about it. I went with only a basic snapshot test, but if
a more detailed test is necessary, I can resubmit.*/

test('Table loads, fetches users, and displays users', () => {
  return store.dispatch(getUsers()).then(() => {
    const app = renderer.create(
      <Provider store={store}>
        <App/>
      </Provider>
    );

    let tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  })
})