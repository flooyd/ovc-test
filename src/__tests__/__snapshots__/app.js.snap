// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table loads, fetches users, and displays users 1`] = `
<div
  className="App"
>
  <table>
    <tbody>
      <tr
        className="tableHeader"
      >
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          City
        </th>
        <th>
          Company
        </th>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <td>
          Leanne Graham
        </td>
        <td>
          Sincere@april.biz
        </td>
        <td>
          Gwenborough
        </td>
        <td>
          Romaguera-Crona
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#eee",
          }
        }
      >
        <td>
          Ervin Howell
        </td>
        <td>
          Shanna@melissa.tv
        </td>
        <td>
          Wisokyburgh
        </td>
        <td>
          Deckow-Crist
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <td>
          Clementine Bauch
        </td>
        <td>
          Nathan@yesenia.net
        </td>
        <td>
          McKenziehaven
        </td>
        <td>
          Romaguera-Jacobson
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#eee",
          }
        }
      >
        <td>
          Patricia Lebsack
        </td>
        <td>
          Julianne.OConner@kory.org
        </td>
        <td>
          South Elvis
        </td>
        <td>
          Robel-Corkery
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <td>
          Chelsey Dietrich
        </td>
        <td>
          Lucio_Hettinger@annie.ca
        </td>
        <td>
          Roscoeview
        </td>
        <td>
          Keebler LLC
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#eee",
          }
        }
      >
        <td>
          Mrs. Dennis Schulist
        </td>
        <td>
          Karley_Dach@jasper.info
        </td>
        <td>
          South Christy
        </td>
        <td>
          Considine-Lockman
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <td>
          Kurtis Weissnat
        </td>
        <td>
          Telly.Hoeger@billy.biz
        </td>
        <td>
          Howemouth
        </td>
        <td>
          Johns Group
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#eee",
          }
        }
      >
        <td>
          Nicholas Runolfsdottir V
        </td>
        <td>
          Sherwood@rosamond.me
        </td>
        <td>
          Aliyaview
        </td>
        <td>
          Abernathy Group
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <td>
          Glenna Reichert
        </td>
        <td>
          Chaim_McDermott@dana.io
        </td>
        <td>
          Bartholomebury
        </td>
        <td>
          Yost and Sons
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#eee",
          }
        }
      >
        <td>
          Clementina DuBuque
        </td>
        <td>
          Rey.Padberg@karina.biz
        </td>
        <td>
          Lebsackbury
        </td>
        <td>
          Hoeger LLC
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
